auth/services/selfsigned/signer.go

```diff
@@ -29,30 +29,62 @@ import (
 	"github.com/nuts-foundation/go-did/vc"
 	"github.com/nuts-foundation/nuts-node/auth/contract"
 	"github.com/nuts-foundation/nuts-node/auth/services"
+	"github.com/nuts-foundation/nuts-node/auth/services/selfsigned/types"
+	"github.com/nuts-foundation/nuts-node/auth/services/selfsigned/web"
+	"github.com/nuts-foundation/nuts-node/core"
+	"github.com/nuts-foundation/nuts-node/vcr"
 	"github.com/nuts-foundation/nuts-node/vcr/credential"
 	"github.com/nuts-foundation/nuts-node/vcr/holder"
 	"github.com/nuts-foundation/nuts-node/vcr/signature/proof"
+	"net/url"
 	"time"
 )
 
 const credentialType = "NutsEmployeeCredential"
 
-func (v service) SigningSessionStatus(ctx context.Context, sessionID string) (contract.SigningSessionResult, error) {
-	s, ok := v.sessions[sessionID]
+// signer implements the contract.Signer interface
+type signer struct {
+	store     types.SessionStore
+	vcr       vcr.VCR
+	publicURL string
+	// The time the user has to sign the contract
+	signingDuration time.Duration
+}
+
+// NewSigner returns an initialized employee identity contract signer
+func NewSigner(vcr vcr.VCR, publicURL string) contract.Signer {
+	return &signer{
+		// NewSessionStore returns an initialized SessionStore
+		store:           NewSessionStore(),
+		vcr:             vcr,
+		publicURL:       publicURL,
+		signingDuration: 10 * time.Minute,
```

Comment:
this feels awfully long to click a button

Reply:
Perhaps, but the first time someone might be wondering what it is, going to call someone, could not find the newly opened tab, etc. So, we can make this shorter but I don't see any security risks in 10 vs 5 minutes.